src/sentry/integrations/github/webhook.py

```diff
@@ -347,18 +347,14 @@ def _handle(self, integration, event, organization, repo, host=None):
                 organization_id=organization.id, external_id=self.get_external_id(user["login"])
             )
         except CommitAuthor.DoesNotExist:
-            try:
-                author = CommitAuthor.objects.get(
-                    organization_id=organization.id, email=author_email
-                )
-            except CommitAuthor.DoesNotExist:
-                author = CommitAuthor.objects.create(
-                    organization_id=organization.id,
-                    email=author_email,
-                    external_id=self.get_external_id(user["login"]),
-                    name=user["login"][:128],
-                )
-
+            author, created = CommitAuthor.objects.get_or_create(
+                organization_id=organization.id,
+                email=author_email,
+                defaults={"name": user["login"][:128]},
+            )
+            if created:
+                author.external_id = self.get_external_id(user["login"])
```

Comment:
Why can't we pass `external_id` to `defaults`?

Reply:
I thought about that, and wasn't sure if there was a possibility of overwriting the `external_id` but it kind of looks like it's always that format? (using the `github:<login_name>`)

Reply:
The `defaults` are only used in the "create" branch so it should be safe to add it.

Reply:
ah yeah true true, updated